### PR TITLE
Use a proper query library to fetch data instead of 🤢🤢🤮🤮

### PR DESCRIPTION
**Pull request by tobycm**
xdddddd too much states

### CODE REVIEW

File: src/components/ListButton.tsx
Comment (pdt1806) [inline src/components/ListButton.tsx:1]: wrap the Link around the product info only, not the whole thing

File: src/pages/Catalog/index.tsx
Comment (pdt1806) [inline src/pages/Catalog/index.tsx:1]: "not unique" you guys sell products with the same name?

jk yea use ids instead

**Review by pdt1806 — APPROVED** [review]
let's gooo nice syntax fixes

that said, consider my comment before merging

File: src/components/ListButton.tsx
Comment (tobycm) [reply src/components/ListButton.tsx:1]: u did this bro xdd

but yeah sure

File: src/pages/Catalog/index.tsx
Comment (tobycm) [reply src/pages/Catalog/index.tsx:1]: these are categories/colors/brands name 😬

File: src/components/ListButton.tsx
Comment (pdt1806) [reply src/components/ListButton.tsx:1]: oh no ;-;

File: src/pages/Catalog/index.tsx
Comment (pdt1806) [reply src/pages/Catalog/index.tsx:1]: my point still stands smh

Comment (tobycm) [reply src/pages/Catalog/index.tsx:1]: i was agreeing xd

Comment (pdt1806) [reply src/pages/Catalog/index.tsx:1]: just merge it already ;-;

**Pull request merged**